src/Provider.js

```diff
@@ -1,42 +1,37 @@
-import { Children, Component, createContext, createElement } from "react"
+import React, { createContext, useContext, useRef } from "react"
 import { shallowEqual } from "./utils/utils"
 
 const specialReactKeys = { children: true, key: true, ref: true }
 
 export const MobXProviderContext = createContext({})
 
-export class Provider extends Component {
-    static contextType = MobXProviderContext
-    static displayName = "MobXProvider"
-
-    constructor(props, context) {
-        super(props, context)
-        this.state = {
-            ...context,
+export function Provider(props) {
+    const parentValue = useContext(MobXProviderContext)
+    const valueRef = useRef(null)
+    if (valueRef.current === null) {
```

Comment:
In light of my other comment, the `useMemo` might work better here, although there is a risk of React throwing it away randomly. So perhaps `useEffect` to update the ref when either parent or value here changes?